### PR TITLE
ENT-3114 | update offer_data only if there is any change.

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**
While updating enterprise coupon
update offer_data only if there is any change related to the offer.
moved validation to the serializer.

### CODE REVIEW

**Review by hasnain-naveed — APPROVED** [review]

File: ecommerce/extensions/api/v2/views/enterprise.py [hunk 1]
```diff
@@ -270,6 +269,24 @@ def update_range_data(self, request_data, vouchers):
         # Since enterprise coupons do not have ranges, we bypass the range update logic entirely.
         pass
 
+    def is_offer_data_updated(self, benefit_value, enterprise_customer, enterprise_catalog, max_uses, email_domains):
+        """ Compares request data with the old coupon data to that there is any change related to offers."""
+        old_data = self.get_serializer(self.get_object()).data
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/api/v2/views/enterprise.py]: old -> existing

File: ecommerce/extensions/api/v2/views/enterprise.py [hunk 2]
```diff
@@ -270,6 +269,24 @@ def update_range_data(self, request_data, vouchers):
         # Since enterprise coupons do not have ranges, we bypass the range update logic entirely.
         pass
 
+    def is_offer_data_updated(self, benefit_value, enterprise_customer, enterprise_catalog, max_uses, email_domains):
+        """ Compares request data with the old coupon data to that there is any change related to offers."""
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/api/v2/views/enterprise.py]: """ Compares request data with the existing coupon data to determine if the offer in voucher needs to be updated."""

File: ecommerce/extensions/api/v2/views/enterprise.py [hunk 3]
```diff
@@ -284,23 +301,13 @@ def update_offer_data(self, request_data, vouchers, site):
...
-            try:
-                max_uses = int(max_uses)
-                if max_uses < 1:
-                    raise ValueError
-            except ValueError:
-                raise ValidationError('max_global_applications field must be a positive number.')
+        if not self.is_offer_data_updated(
+                benefit_value, enterprise_customer, enterprise_catalog, max_uses, email_domains
+        ):
+            return
```
Comment (irfanuddinahmad) [inline ecommerce/extensions/api/v2/views/enterprise.py:309]: Comment: Offer data does not need to be updated for current request

**Review by irfanuddinahmad — APPROVED** [review]
Some small changes ... otherwise looks good!

**Review by moconnell1453 — APPROVED** [review]

**Pull request merged**